### PR TITLE
refactor(nodebuilder/p2p): Allow specifying network alias with `--p2p.network` flag

### PR DESCRIPTION
**Pull request by renaynay**
* Allows users to specify the network alias rather than the actual value of the network ID when passing `--p2p.network`
* Fails out on init if invalid network specified via `--p2p.network`

Supersedes https://github.com/celestiaorg/celestia-node/pull/1353

Ensures https://github.com/celestiaorg/docs/pull/304 does not need to happen.

### CODE REVIEW

File: nodebuilder/p2p/flags.go [hunk 1]
```diff
@@ -67,17 +67,26 @@ func ParseFlags(
...
+	// check if user provided an alias
+	parsedNetwork, ok := networkAliases[parsed]
+	if !ok {
+		// check if user provided the actual network value
+		if err := Network(parsed).Validate(); err == nil {
+			return Network(parsed), nil
+		}
+		return "", fmt.Errorf("invalid network specified: %s", parsed)
+	}
+	return parsedNetwork, nil
```
Comment (Wondertan) [inline nodebuilder/p2p/flags.go:89]: style nit: reduce nesting by doing `if ok`

**Review by Wondertan — DISMISSED** [review]

**Review by distractedm1nd — DISMISSED** [review]
nice

File: nodebuilder/p2p/flags.go (hunk 1 above)
Comment (renaynay) [reply nodebuilder/p2p/flags.go:89]: whoops, u right

**Review by distractedm1nd — APPROVED** [review]
looks good but please create issue for writing unit tests for all of the network parsing functionality (not this PR though)

File: nodebuilder/p2p/network.go [hunk 2]
```diff
@@ -46,6 +46,15 @@ var networksList = map[Network]struct{}{
 	Private: {},
 }
 
+// networkAliases is a strict list of all known long-standing networks
+// mapped from the string representation of their *alias* (rather than
+// their actual value) to the Network.
+var networkAliases = map[string]Network{
+	"arabica": Arabica,
+	"mamaki":  Mamaki,
+	"private": Private,
```
Comment (walldiss) [inline nodebuilder/p2p/network.go:55]: Nit optional suggestion to make `Name() string`  method for `Network` type with static value.

**Review by walldiss — APPROVED** [review]
Looks good

**Review by walldiss — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**